### PR TITLE
DRAFT: feat(react-nav-preview): add transition examples and fix some Drawer compatibility issues

### PR DESCRIPTION
**Pull request by marcosmoura**

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">38</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Avatar-mount_base.html">653</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Avatar-mount.html">656</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Button-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Button-mount.html">321</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Field-mount_base.html">1157</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Field-mount.html">1167</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProvider-mount_base.html">724</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProvider-mount.html">753</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-mount_base.html">85</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-mount.html">88</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">38</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/MakeStyles-mount_base.html">958</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/MakeStyles-mount.html">974</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Persona-mount_base.html">1855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/Persona-mount.html">1791</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/SpinButton-mount_base.html">1470</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/SpinButton-mount.html">1463</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/SwatchPicker-mount_base.html">1573</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31657/perf-test-react-components/SwatchPicker-mount.html">1633</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-drawer-64e63a1b-c3e5-46f2-bf67-50fcf940da74.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-drawer-64e63a1b-c3e5-46f2-bf67-50fcf940da74.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31657&commitId=4308055ad1e5bc7febd82956965d90be59418332&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|0| Added|

</details>


  <details><summary><b>Card Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged.size.normal.chromium.png|0| Added|

</details>


  <details><summary><b>Card Converged - Interactive</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Interactive.appearance interactive - Filled Alternative.click.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Filled Alternative.hover.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Filled Alternative.normal.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.click.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.hover.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative - High Contrast.normal.chromium.png|0| Removed|

</details>


  <details><summary><b>SwatchPicker Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.default.default.chromium.png|0| Added|
|SwatchPicker Converged.Spacing.default.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by mltejera** [comment]
Shift is rewriting a bunch of the animation work on Drawer, so going to close this and draft of his work once he finishes.

**Pull request closed without merging**